plugins/poll/assets/javascripts/discourse/components/poll-results-pie.gjs

```diff
@@ -0,0 +1,129 @@
+import { PIE_CHART_TYPE } from "../components/modal/poll-ui-builder";
+
+export default class PollResultsPieComponent extends Component {
+  htmlLegendPlugin = {
+    id: "htmlLegend",
+
+    afterUpdate(chart, args, options) {
+      const ul = document.getElementById(options.containerID);
+      ul.innerHTML = "";
+
+      const items = chart.options.plugins.legend.labels.generateLabels(chart);
+      items.forEach((item) => {
+        const li = document.createElement("li");
+        li.classList.add("legend");
+        li.onclick = () => {
+          chart.toggleDataVisibility(item.index);
+          chart.update();
+        };
+
+        const boxSpan = document.createElement("span");
+        boxSpan.classList.add("swatch");
+        boxSpan.style.background = item.fillStyle;
+
+        const textContainer = document.createElement("span");
+        textContainer.style.color = item.fontColor;
+        textContainer.innerHTML = item.text;
+
+        if (!chart.getDataVisibility(item.index)) {
+          li.style.opacity = 0.2;
+        } else {
+          li.style.opacity = 1.0;
+        }
+
+        li.appendChild(boxSpan);
+        li.appendChild(textContainer);
+
+        ul.appendChild(li);
+      });
+    },
+  };
+
+  stripHtml = (html) => {
+    let doc = new DOMParser().parseFromString(html, "text/html");
+    return doc.body.textContent || "";
+  };
+
+  pieChartConfig = (data, labels, opts = {}) => {
+    const aspectRatio = "aspectRatio" in opts ? opts.aspectRatio : 2.2;
+    const strippedLabels = labels.map((l) => this.stripHtml(l));
+
+    return {
+      type: PIE_CHART_TYPE,
+      data: {
+        datasets: [
+          {
+            data,
+            backgroundColor: getColors(data.length),
+          },
+        ],
+        labels: strippedLabels,
+      },
+      plugins: [this.htmlLegendPlugin],
+      options: {
+        responsive: true,
+        aspectRatio,
+        animation: { duration: 0 },
+        plugins: {
+          legend: {
+            labels: {
+              generateLabels() {
+                return labels.map((text, index) => {
+                  return {
+                    fillStyle: getColors(data.length)[index],
+                    text,
+                    index,
+                  };
+                });
+              },
+            },
+            display: false,
+          },
+          htmlLegend: {
+            containerID: opts?.legendContainerId,
+          },
+        },
+      },
+    };
+  };
+
+  get canvasId() {
+    return htmlSafe(`poll-results-chart-${this.args.id}`);
+  }
+
+  get legendId() {
+    return htmlSafe(`poll-results-legend-${this.args.id}`);
+  }
+
+  @action
+  drawPie() {
+    loadScript("/javascripts/Chart.min.js").then(() => {
+      const data = this.args.options.mapBy("votes");
+      const labels = this.args.options.mapBy("html");
+      const config = this.pieChartConfig(data, labels, {
+        legendContainerId: `poll-results-legend-${this.args.id}`,
```

Comment:
should use `this.legendId` ?

Although, instead of doing this id dance, I would recommend {{this.registerLegentElement}} on the <ul> and then having a modifier

registerLegentElement = modifier((element) => {
  this.legendElement = element;
})

Reply:
So @jjaffeux I've made the corresponding suggested changes here, applying this also to the ContainerID: https://github.com/discourse/discourse/pull/27155/commits/3e1ea6cbbbef605399cf8c04a777e0f5cc2115c8

* Not entirely convinced we don't end up with a "parallel" pair of strategies now?
* We still have an intermittent FE test failure on this component, which is a bit frustrating.  The pie chart code is _exactly_ the same here as it was prior to the Glimmer migration, yet this didn't seem to have been an issue before.  Any suggestions on how I might make this more reliable?


```
Error: QUnit Test Failure: Global error: Uncaught TypeError: Cannot set properties of null (setting 'innerHTML') at http://localhost:7357/assets/plugins/poll.js, line 1706
 While executing test: Poll | Component | poll-results-standard: options in ascending order

not ok 591 Chrome 120.0 - [undefined ms] - Global error: Uncaught TypeError: Cannot set properties of null (setting 'innerHTML') at http://localhost:7357/assets/plugins/poll.js, line 1706
Testem finished with non-zero exit code. Tests failed.

 While executing test: Poll | Component | poll-results-standard: options in ascending order
    ---
        browser log: |
            {"type":"error","text":"Uncaught TypeError: Cannot set properties of null (setting 'innerHTML') at http://localhost:7357/assets/plugins/poll.js, line 1706\n","testContext":{"id":211,"name":"Poll | Component | poll-results-standard: options in ascending order","items":[],"state":"complete"}}
    ...
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.

Tests Failed
Error: Process completed with exit code 1.
```

Reply:
@jjaffeux as you know we should have resolved this as part of #27765